native/objcexport-header-generator/impl/analysis-api/src/org/jetbrains/kotlin/objcexport/getNSEnumType.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2010-2025 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.objcexport
+
+import org.jetbrains.kotlin.analysis.api.annotations.KaAnnotationValue
+import org.jetbrains.kotlin.analysis.api.symbols.KaClassSymbol
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.name.FqName
+
+/** Returns the NSEnum type for the given enum type if the corresponding annotation is set; null otherwise */
+fun ObjCExportContext.getNSEnumType(symbol: KaClassSymbol): String? {
+    val classId = ClassId(FqName("kotlin.native"), FqName("ObjCEnum"), false)
+    val annotation = symbol.annotations[classId].firstOrNull()
+    return if (annotation == null) {
+        null
+    } else if (annotation.arguments.isEmpty()) {
+        getObjCClassOrProtocolName(symbol).objCName + "NSEnum"
+    } else {
+        (annotation.arguments[0].expression as KaAnnotationValue.ConstantValue).value.value as String
```

Comment:
`resolveStringConstantValue` can be reused instead.

https://github.com/JetBrains/kotlin/blob/a4fbe41f7688185abf770706a923b44d20ada15b/native/objcexport-header-generator/impl/analysis-api/src/org/jetbrains/kotlin/objcexport/resolveObjCNameAnnotation.kt#L50